### PR TITLE
[MO] - [system test] -> kafka connect with scram-sha plain

### PR DESCRIPTION
**Pull request by Seequick1**
Signed-off-by: Seequick1 <morsak@redhat.com>

### Type of change

- Enhancement / new feature

### Description

This PR adding a new test case for Kafka connect using Scram-sha authentification with plain communication.

### Checklist

- [x] Write tests
- [ ] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java [hunk 1]
```diff
@@ -157,6 +161,77 @@ void testKafkaConnectWithFileSinkPlugin() throws Exception {
         KafkaTopicUtils.waitForKafkaTopicDeletion(CONNECT_TOPIC_NAME);
     }
 
+    @Test
+    @Tag(NODEPORT_SUPPORTED)
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: what does it mean? sorry for the stupid question :-)

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java [hunk 2]
```diff
@@ -157,6 +161,77 @@ void testKafkaConnectWithFileSinkPlugin() throws Exception {
...
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withNewPlain()
+                                .withNewKafkaListenerAuthenticationScramSha512Auth()
+                                .endKafkaListenerAuthenticationScramSha512Auth()
+                            .endPlain()
+                            .withNewKafkaListenerExternalNodePort()
+                                .withNewKafkaListenerAuthenticationScramSha512Auth()
+                                .endKafkaListenerAuthenticationScramSha512Auth()
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: why configuring the external node port? I don't see client or kafka connect are going to use it. Maybe it's related to the stupid question above :-)

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java [hunk 3]
```diff
@@ -364,7 +439,7 @@ void testSecretsWithKafkaConnectWithTlsAndTlsClientAuthentication() throws Excep
 
     @Test
     @Tag(NODEPORT_SUPPORTED)
-    void testSecretsWithKafkaConnectWithTlsAndScramShaAuthentication() throws Exception {
+        void testSecretsWithKafkaConnectWithTlsAndScramShaAuthentication() throws Exception {
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: I think you added this tab my mistake.

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java (hunk 1 above)
Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: The Nodeport supported means that the only environments which support Nodeports for instance in AWS nodeport is working only with specific configuration the same as for load balancers in OCP4. So, with this notation, we are excluding some tests to run only in a specific environment.

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java (hunk 2 above)
Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: If you take a look at  `line 223` you will see the invocation of the client's method 

```
kafkaClient.sendAndRecvMessagesScramSha(USER_NAME, NAMESPACE, CLUSTER_NAME, CONNECT_TOPIC_NAME, 2);
```
which means, that Kafka Connect communicates with internal listener of Kafka and this client communicates with the external listener.

I hope, this makes any sense  :100: .

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java (hunk 3 above)
Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: Totally right. Thanks

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java (hunk 2 above)
Comment (ppatierno) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: Sorry but how it's clear that client is using external listener from that call? There is a username, a namespace, a cluster name and a topic name :-/

Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: I have got an idea but if you took a closer look on from where the client is you will find out that he is inside `externalClients package`. Anyway i can rename his for instance `externalBasicKafkaClient`. What do you think? @ppatierno

**Comment by Frawless** [comment]
Cherry-pick to `release-0.17.x` please

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java (hunk 2 above)
Comment (ppatierno) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: Yeah, I think that any solution which makes clear from where the client is connecting works fine for me.

Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: Done.

**Comment by scholzj** [comment]
@Seequick1 Could you please resovle the conflicts?

**Pull request merged**